portfolio/src/main/java/com/google/sps/data/BlogHashMap.java

```diff
@@ -147,39 +104,32 @@ public void putInMap(List<BlogMessage> messages) {
     return values;
   } 
 
-  // Takes a list of tags, and
-  // Returns a linkedList of all values of BlogMessage type for the specified tags.
-  public LinkedList<BlogMessage> getMessages(List<String> tags) {
-    LinkedList<BlogMessage> values = new LinkedList<BlogMessage>();
-    for (String tag : tags){
-      if (map.containsKey(tag)) {
-        values.addAll(map.get(tag));
-      } else {
-        System.err.println("Tag not found. Tag : " + tag);
-      }    
-    }
-    return values;
-  }
-
-  // Returns a requested amount of all BlogMessages for specified tags.
+  // This method takes a list of tags and a number of messages to load, and
+  // Returns a requested (or default) amount of BlogMessages under specified tag/tags.
+  // If tag/tags are not specified, a requested (or default) amount of all BlogMessages will be returned.
   public LinkedList<BlogMessage> getMessages(List<String> tags, int limit) {
-    if (limit < 0) {
-      return new LinkedList<>();
+    if (limit <= 0) {
+      limit = DEFAULT_LOAD_AMOUNT;
+    }
+    if (tags.isEmpty()) {
+      return getMessages(limit);
+    }
+    if (tags.size() == 1) {
+      return getMessages(tags.get(0), limit);
     }
 
     LinkedList values = new LinkedList();
-
     for (String tag : tags) {
-      if (limit <= 0) {break;}
-      Iterator iterator;
+      if (limit <= 0) 
+      {break;}
```

Comment:
Nit: use the recommended style guide for consistency:

if (limit <= 0) {
  break;
}